File: weather.js
const APIURL = "https://samples.openweathermap.org/data/2.5/weather?q=Lyon,fr&appid=b6907d289e10d714a6e88b30761fae22"

function getWeather(callback) {
  const Http = new XMLHttpRequest()
  Http.open("GET", APIURL)
  Http.send()
  Http.onreadystatechange=(e)=>{
    console.log(e)
    callback(Http.responseText)
  }
}

File: weather.py
import time
import requests

API = "https://api.openweathermap.org/data/2.5/weather?appid=77d4b14c8aa6e7d25da6b2aebe4c33b7&q=Lyon&units=metric&lang=fr"

while True:

    r = requests.get(API).text
    with open("weather.json", "w+") as file:
        file.write(r)

    time.sleep(60)
